src/main/java/org/opensearch/security/ssl/util/SSLRequestHelper.java

```diff
@@ -121,25 +119,14 @@ public String toString() {
     public static SSLInfo getSSLInfo(
         final Settings settings,
         final Path configPath,
-        final RestRequest request,
+        final SecurityRequest request,
         PrincipalExtractor principalExtractor
     ) throws SSLPeerUnverifiedException {
-
-        if (request == null || request.getHttpChannel() == null || !(request.getHttpChannel() instanceof Netty4HttpChannel)) {
-            return null;
-        }
-
-        final Netty4HttpChannel httpChannel = (Netty4HttpChannel) request.getHttpChannel();
-        SslHandler sslhandler = (SslHandler) httpChannel.getNettyChannel().pipeline().get("ssl_http");
-        if (sslhandler == null && httpChannel.inboundPipeline() != null) {
-            sslhandler = (SslHandler) httpChannel.inboundPipeline().get("ssl_http");
-        }
-
-        if (sslhandler == null) {
+        final SSLEngine engine = request.getSSLEngine();
```

Comment:
nit: sslEngine?

Reply:
I'll keep it as is, thanks for the feedback